src/libespeak-ng/encoding.c

```diff
@@ -562,6 +562,8 @@ string_decoder_getc_utf_8(espeak_ng_TEXT_DECODER *decoder)
 		ret = (ret << 6) + (c & 0x3F);
 		if (((c = *decoder->current++) & LEADING_2_BITS) != UTF8_TAIL_BITS) goto error;
 		ret = (ret << 6) + (c & 0x3F);
+		// avoid the "I umlaut a half" bug
+		if (ret == 0xFFFD) goto substitute;
```

Comment:
Why just just immediately return 0x001A; ?

Reply:
Hi @sthibaul 

This new `if` just replaces the `ret` value, which would be `0xFFFD`, with the value `0x001A`.

When `string_decoder_getc_auto()` catches the return value `0x001A`, it changes back again to `0xFFFD`.

Not very elegant, but it's a workaround.

Reply:
I used a `goto` to make the workaround more explicit than just returning `0x001A`.

Reply:
In fact, it could be fixed with just one if:

```c
// avoid the "I umlaut a half" bug
if (ret == 0xFFFD) return 0x001A;
```

Reply:
Changed the PR to immediately return `0x001A`. Only one line of code added.